### PR TITLE
front-end

### PR DESCRIPTION
**Pull request by teluashish**

### CODE REVIEW

File: src/views/Actors.vue [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<template>
+<div>
+    <PersonFormModal :isActor="true"> </PersonFormModal> 
+    <br><br>
+    <div align="center" id="card-body">
+    <Row :gutter="20" :type="'flex'">
+        <CardComponent v-for="(actor,idx) in getActors" :key="idx" :actor="actor" :isMovieCard="false" :isActor="true"></CardComponent>
```
Comment (adatdeltax) [inline src/views/Actors.vue:7]: Please rename idx to something meaningful like id or index

File: src/views/Movies.vue [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+<template>
+<div>
+    <br>
+    <FormModal :isEdit="false" ></FormModal>
+    <br><br>
+    <br>
+    <div align="center">
+    <Row :gutter="20" :type="'flex'">
+        <CardComponent v-for="(movie,idx) in getMovies" :key="idx" :movie="movie" :isMovieCard="true"></CardComponent>
+    </Row>
```
Comment (adatdeltax) [inline src/views/Movies.vue:11]: Please rename idx to something meaningful like id or index

File: src/views/Producers.vue [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<template>
+<div>     
+    <PersonFormModal :isActor="false"> </PersonFormModal> 
+    <br><br>
+    <Row :gutter="20" :type="'flex'">
+        <CardComponent v-for="(producer,idx) in getProducers" :key="idx" :producer="producer" :isMovieCard="false" :isActor="false"></CardComponent>
+    </Row>
+ </div>
```
Comment (adatdeltax) [inline src/views/Producers.vue:8]: Please rename idx to something meaningful like id or index

File: src/store/store.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        },
+
+        //Producers
+        async getProducersAsync(context){
+            await axios.get('/producers').then( producers => 
+                context.commit('assignProducers',producers['data'])).catch( error =>
+                    window.alert('producers fetch: '+error)
+                )
+        },
+        async postProducerAsync(context, producer){
```
Comment (adatdeltax) [inline src/store/store.js:149]: - Please give relatable method name , with the action syntax it is known that it is async method i.e action name can be as simple as getProducers, getActors. 

- Please remove unnecessary console logs

File: src/components/CardComponent.vue [hunk 5]
```diff
@@ -0,0 +1,28 @@
+<template>
+    <Col :xs="20" :sm="16" :md="12" :lg="8" :xl="6">
+        <div id = "cardComponent" align="center">
+            <MovieDetailsComponent v-if = "isMovieCard" :movie="movie" />
+            <PersonDetailsComponent v-if = "!isMovieCard" :person = "isActor ? actor : producer" :isActor="isActor" />
```
Comment (adatdeltax) [inline src/components/CardComponent.vue:5]: - Please rename component names as it is known by default it is component , component names can be changed from MovieDetailsComponent --> MovieDetails or Movie, PersonDetailsComponent --> PersonDetails or Person

File: src/main.js [hunk 6]
```diff
@@ -1,8 +1,24 @@
...
+// import style
+import 'view-design/dist/styles/iview.css';
+import {store} from './store/store';
+
+
+Vue.use(ViewUI);
+
+axios.defaults.baseURL = "https://localhost:64536/";
 Vue.config.productionTip = false
+export const eventBus = new Vue();
```
Comment (adatdeltax) [inline src/main.js:17]: I don't think eventBus is used anywhere, please remove it.

File: src/components/FormModal.vue [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+
+        },
+        methods: {
+            ...mapActions(['postMovieAsync','putMovieAsync','getGenresAsync','getMoviesAsync']),
+            submit () {
+                console.log("submit")
+                this.movieModal = false
+                this.modify()
+                if(!this.isEdit)this.add()
+                else this.edit(this.movie.id)
```
Comment (adatdeltax) [inline src/components/FormModal.vue:142]: Please remove console logs

File: src/components/MovieDetailsComponent.vue [hunk 8]
```diff
@@ -0,0 +1,61 @@
+<template>
+    <div id="movieDetailsComponent">
+                <Card :bordered="false">
+                    <p slot="title"> {{movie.name}} </p>
+                    <img :src="movie.coverImage" /> <br /><br />
+                    <div align="left">
+                    <p><b>Year of release: </b> {{movie.year}}</p>
+                    <p><b>Actors: </b> <span v-for="(actor,idx) in movie.actors" :key="actor.id"> {{actor.name}}<span v-if="idx+1 < movie.actors.length">, </span></span></p>
+                    <p><b>Genres: </b> <span v-for="(genre,idx) in movie.genres" :key="genre.id"> {{genre.name}}<span v-if="idx+1 < movie.genres.length">, </span> </span></p>
```
Comment (adatdeltax) [inline src/components/MovieDetailsComponent.vue:9]: again here pls change idx to a meaningful name

File: src/components/CardComponent.vue [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        </div>
+    </Col>
+</template>
```
Comment (adatdeltax) [inline src/components/CardComponent.vue:8]: isActor ? actor : producer can this be a computed

File: src/components/CardComponent.vue (hunk 5 above)
Comment (adatdeltax) [reply src/components/CardComponent.vue:5]: I think simple if else is more than enough here instead of two if blocks

**Review by adatdeltax — CHANGES_REQUESTED** [review]

File: src/components/CardComponent.vue (hunk 9 above)
Comment (teluashish) [reply src/components/CardComponent.vue:8]: yes, it can be.

**Pull request merged**